packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart

```diff
@@ -138,3 +142,51 @@ class LatLngBounds {
   @override
   int get hashCode => Object.hash(southwest, northeast);
 }
+
+/// A data point entry for a heatmap.
+/// This is a geographical data point with a weight attribute.
+class WeightedLatLng extends LatLng {
+  /// Creates a [WeightedLatLng] with the specified [weight]
+  const WeightedLatLng(double latitude, double longitude, {this.weight = 1.0})
+      : super(latitude, longitude);
+
+  /// The weighting value of the data point.
+  final double weight;
+
+  /// Converts this object to something serializable in JSON.
+  @override
+  Object toJson() {
```

Comment:
As with heatmap itself: can we eliminate this and fromJson?

Reply:
Not sure if this is safe to remove. Let me know your thoughts.